parsing/extensions.mli

```diff
@@ -76,72 +79,28 @@ module Immutable_arrays : sig
   val pat_of : loc:Location.t -> pattern -> Parsetree.pattern_desc
 end
 
-(** The module type of language extension ASTs, instantiated once for each
-    syntactic category.  We tend to call the pattern-matching functions here
-    with unusual indentation, not indenting the [None] branch further so as to
-    avoid merge conflicts with upstream. *)
```

Comment:
I think we lost the comments about indentation, and they're very important documentation